### PR TITLE
[CFX-5937] expand unit tests coverage for internal/shell

### PR DESCRIPTION
**Pull request by jpgomes**
# RATIONALE

expand unit tests coverage for internal/shell from 20% to 52.5% 

## CHANGES

extend the coverage from 20% to 52.5%. The new tests cover:
  - ResolveShell — up to 87.5% (the only uncovered line is the DetectShell error return, which requires the test runner to have no detectable shell)
  - parentProcessNameWindows — up to 41.7% (the CSV parsing logic after a successful tasklist call remains unreachable on Linux, as expected)

## PR Automation

**Comment-Commands:** Trigger CI by commenting on the PR:
- `/trigger-smoke-test` or `/trigger-test-smoke` - Run smoke tests
- `/trigger-install-test` or `/trigger-test-install` - Run installation tests

**Labels:** Apply labels to trigger workflows:
- `run-smoke-tests` or `go` - Run smoke tests on demand (only works for non-forked PRs)

> [!IMPORTANT]
> **For Forked PRs:** The `run-smoke-tests` label won't work. A required **Smoke Tests** check will block merge until a maintainer acts:
> - A maintainer uses `/approve-smoke-tests` to run smoke tests (results will set the check)
> - A maintainer uses `/skip-smoke-tests` to bypass the check without running tests
>
> Please comment requesting a maintainer review if you need smoke tests to run.

---

> [!NOTE]
> **Low Risk**
> Low risk: changes are test-only and primarily add table-driven fixtures around shell name normalization and resolution logic, with no production code modifications.
> 
> **Overview**
> Expands `internal/shell` unit tests to cover more `DetectShell`/`ResolveShell` behaviors and edge cases, including canonical `pwsh`→`powershell` normalization and `SHELL` env-var path fixtures.
> 
> Adds assertions for `SupportedShells()` contents, `ResolveShell` passthrough vs auto-detect, and a Windows `tasklist` failure case via an invalid PID to ensure `parentProcessNameWindows` safely returns an empty string.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 9caa5b626bfb919f84fd96749dd7a211e86a49f3. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: internal/shell/shell_test.go [hunk 1]
```diff
@@ -68,3 +69,158 @@ func TestDetectShell_EnvVarFallback(t *testing.T) {
...
+		{shellEnv: "/opt/homebrew/bin/zsh", want: "zsh"},
+		{shellEnv: "/opt/homebrew/bin/fish", want: "fish"},
+		// Bare names — already resolved via PATH in the user's config
+		{shellEnv: "bash", want: "bash"},
+		{shellEnv: "zsh", want: "zsh"},
+		{shellEnv: "fish", want: "fish"},
+		{shellEnv: "sh", want: "sh"},
+		// PowerShell variants — pwsh must normalise to "powershell"
+		{shellEnv: "/usr/local/bin/pwsh", want: "powershell"},
+		{shellEnv: "/opt/homebrew/bin/pwsh", want: "powershell"},
```
Comment (ajalon1) [inline internal/shell/shell_test.go:105]: You can brew install powershell??

**Review by ajalon1 — APPROVED** [review]

File: internal/shell/shell_test.go (hunk 1 above)
Comment (jpgomes) [reply internal/shell/shell_test.go:105]: https://learn.microsoft.com/en-us/powershell/scripting/install/install-powershell-on-macos?view=powershell-7.6 seems so
